x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/VersionHttpResource.java

```diff
@@ -50,22 +52,33 @@ public VersionHttpResource(final String resourceOwnerName, final Version minimum
      * If it does not, then there is nothing that can be done except wait until it does. There is no publishing aspect to this operation.
      */
     @Override
-    protected boolean doCheckAndPublish(final RestClient client) {
+    protected void doCheckAndPublish(final RestClient client, final ActionListener<Boolean> listener) {
         logger.trace("checking [{}] to ensure that it supports the minimum version [{}]", resourceOwnerName, minimumVersion);
 
-        try {
-            Request request = new Request("GET", "/");
-            request.addParameter("filter_path", "version.number");
-            return validateVersion(client.performRequest(request));
-        } catch (IOException | RuntimeException e) {
-            logger.error(
-                    (Supplier<?>)() ->
-                        new ParameterizedMessage("failed to verify minimum version [{}] on the [{}] monitoring cluster",
-                                                 minimumVersion, resourceOwnerName),
-                    e);
-        }
+        final Request request = new Request("GET", "/");
+        request.addParameter("filter_path", "version.number");
+
+        client.performRequestAsync(request, new ResponseListener() {
+            @Override
+            public void onSuccess(final Response response) {
+                try {
+                    // malformed responses can cause exceptions during validation
+                    listener.onResponse(validateVersion(response));
+                } catch (IOException | RuntimeException e) {
```

Comment:
maybe just catch `Exception` here.

Reply:
Kind of a shame that `ResponseListener::onResponse` doesn't allow that method to throw an `Exception` because it _catches_ `Exception` and passes it to `onFailure` already.

Reply:
yes, that would avoid such try-catch claise here.